GoogleUtilities/Environment/GULHeartbeatDateStorage.m

```diff
@@ -18,14 +18,44 @@
 #import "GoogleUtilities/Environment/Public/GoogleUtilities/GULSecureCoding.h"
 
 @interface GULHeartbeatDateStorage ()
+#if TARGET_OS_TV
```

Comment:
It seems there are not too much shared code (if any?) between the file based and user defaults based implementation. Would it be cleaner and more readable to have separate classes for them? These classes with different implementation may implement a protocol or be inherited from the base abstract GULHeartbeatDateStorage. In this case it will be up to client (FIRHeartbeatInfo in fact) to instantiate proper implementation.

I'm not too strong about it now, but I see more benefits of splitting the implementation eventually (maybe in the following PRs/releases).